data/test/scenarios/wml_tests/FilterWML/special_id_active_lua_function.cfg

```diff
@@ -0,0 +1,106 @@
+#####
+# This unit test reproduces setup of weapon special such as https://github.com/inferno8/wesnoth-Era_of_Magic/blob/adf227cba2e66602c4bb49dc203fa751865aa124/utils/beam.cfg, which is not supposed to work in rpg.
+##
+# API(s) being tested: [event][filter_attack]special_id_active, weapon special [filter_self]lua_function
+##
+# Expected end state for test case 1: attacker hits event is triggered twice according to number of Elvish Archer (alice) strikes
+# Expected end state for test case 2: attacker hits event is not triggered since weapon special is not active anymore
+#####
+
+{COMMON_KEEP_A_B_UNIT_TEST "special_id_active_lua_function" (
+    [event]
+        name=preload
+        first_time_only=yes
+        [lua]
+            code=<<
+conditional_special_active = function()
+	return wml.variables.conditional_special_active
```

Comment:
Having the same name for the Lua function and the WML variable makes the test harder to read. I suggest renaming the WML variable as `allow_extra_damage`.

Also, the indentation is off here (and wmlindent seems happy both as-is and when it's corrected).

Reply:
It is Lua, out of scope of wmlindent.

Reply:
Yes, but the coding standard that's at least de-facto for the data/test directory is to have `>>` at the same level of indentation as `code=<<`, and the first line of the Lua being one indent more than that.

Reply:
I wouldnt mix Lua and wml indents.

I found one example of my preferred style
```
data/campaigns/Son_Of_The_Black_Eye/utils/utils.cfg-    [lua]
data/campaigns/Son_Of_The_Black_Eye/utils/utils.cfg:        code = <<
data/campaigns/Son_Of_The_Black_Eye/utils/utils.cfg-function wesnoth.custom_synced_commands.ship_unload(cfg)
```

Reply:
I would accept compromise as code=<<conditional_special_active = function() return wml.variables.allow_extra_damage end>>